components/eam/cime_config/buildnml

```diff
@@ -90,6 +90,7 @@ def buildnml(case, caseroot, compname):
         comp = ""
         if comp_interface == "mct":  comp = "mct"
         if comp_interface == "esmf": comp = "esmf"
```

Comment:
This seems like a redundant assignment. if `comp_interface` is non-null string, then `comp = comp_interface` ?